src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -52,51 +53,68 @@ public class RedisHubLifetimeManager<THub> : HubLifetimeManager<THub>, IDisposab
 
-            var writer = new LoggerTextWriter(logger);
             RedisLog.ConnectingToEndpoints(_logger, options.Value.Options.EndPoints, _serverName);
-            _redisServerConnection = _options.Connect(writer);
+        }
 
-            _redisServerConnection.ConnectionRestored += (_, e) =>
+        private async Task EnsureRedisServerConnection()
+        {
+            if (_redisServerConnection == null)
             {
-                // We use the subscription connection type
-                // Ignore messages from the interactive connection (avoids duplicates)
-                if (e.ConnectionType == ConnectionType.Interactive)
+                try
                 {
-                    return;
-                }
+                    await _connectionLock.WaitAsync();
+                    if (_redisServerConnection == null)
+                    {
+                        var writer = new LoggerTextWriter(_logger);
+                        _redisServerConnection = await _options.Connect(writer);
+                        _bus = _redisServerConnection.GetSubscriber();
+                        _redisServerConnection.ConnectionRestored += (_, e) =>
+                        {
+                        // We use the subscription connection type
+                        // Ignore messages from the interactive connection (avoids duplicates)
+                        if (e.ConnectionType == ConnectionType.Interactive)
+                            {
+                                return;
+                            }
 
-                RedisLog.ConnectionRestored(_logger);
-            };
+                            RedisLog.ConnectionRestored(_logger);
+                        };
 
-            _redisServerConnection.ConnectionFailed += (_, e) =>
-            {
-                // We use the subscription connection type
-                // Ignore messages from the interactive connection (avoids duplicates)
-                if (e.ConnectionType == ConnectionType.Interactive)
+                        _redisServerConnection.ConnectionFailed += (_, e) =>
+                        {
+                        // We use the subscription connection type
+                        // Ignore messages from the interactive connection (avoids duplicates)
+                        if (e.ConnectionType == ConnectionType.Interactive)
+                            {
+                                return;
+                            }
+
+                            RedisLog.ConnectionFailed(_logger, e.Exception);
+                        };
+                        if (_redisServerConnection.IsConnected)
+                        {
+                            RedisLog.Connected(_logger);
+                        }
+                        else
+                        {
+                            RedisLog.NotConnected(_logger);
+                        }
+
+                        SubscribeToAll();
```

Comment:
??? We only need to subscribe once, the library handles resubscribing on connection restores

Reply:
Also, you aren't guaranteed to set this callback before you connect to redis so you would miss setting up all the subscriptions

Reply:
> We only need to subscribe once, the library handles resubscribing on connection restores

Okay. This is how it was implemented before. I'll fix it.

Reply:
Make sure to verify that actually works. It's possible the API won't allow you to subscribe until after connecting

Reply:
@BrennanConroy I think you're confusing this because of indentation. This logic only happens on initial startup.

@mikaelm12 Fix the indentation :). If this is the same as the initial code, just leave it in this PR.

Reply:
Turns out formatting is important 😄